packages/headless/src/controllers/generated-answer/headless-generated-answer.ts

```diff
@@ -30,11 +31,17 @@ export function buildGeneratedAnswer(
   engine: SearchEngine,
   props: GeneratedAnswerProps = {}
 ): GeneratedAnswer {
-  const controller = buildSearchAPIGeneratedAnswer(
-    engine,
-    generatedAnswerAnalyticsClient,
-    props
-  );
+  const controller = props.answerConfigurationId
+    ? buildKnowledgeGeneratedAnswer(
+        engine,
+        generatedAnswerAnalyticsClient,
+        props
+      )
+    : buildSearchAPIGeneratedAnswer(
+        engine as SearchEngine,
+        generatedAnswerAnalyticsClient,
+        props
+      );
```

Comment:
Why are we not having such a logic for the insight generated answer controller?
is the feature of the new `buildKnowledgeGeneratedAnswer` controller not yet supported for insight?

Reply:
This was not in the radar for the MVP and for the sake of being able to give the feature to some MVP customers, we only support the basic interface usecase for now.  No doubt we will have to support insight at some point though.